modules/simple_forms_api/app/form_mappings/vba_21_0966.json.erb

```diff
@@ -1,66 +1,66 @@
-  "F[0].Page_1[0].Mailing_Address_ZIPOrPostalCode_FirstFiveNumbers[0]": "<%= data.dig('surviving_dependent_mailing_address', 'postal_code')&.[](0..4) %>",
-  "F[0].Page_1[0].Mailing_Address_Country[0]": "<%= data.dig('surviving_dependent_mailing_address', 'country') %>",
-  "F[0].Page_1[0].Mailing_Address_StateOrProvince[0]": "<%= data.dig('surviving_dependent_mailing_address', 'state') %>",
-  "F[0].Page_1[0].Mailing_Address_City[0]": "<%= data.dig('surviving_dependent_mailing_address', 'city') %>",
-  "F[0].Page_1[0].Mailing_Address_ApartmentOrUnitNumber[1]": "<%= data.dig('surviving_dependent_mailing_address', 'street2') %>",
-  "F[0].Page_1[0].Mailing_Address_NumberAndStreet[1]": "<%= data.dig('surviving_dependent_mailing_address', 'street') %>",
-  "F[0].Page_1[0].ClaimantsInternational_Phone_Number[0]": "<%= data['surviving_dependent_international_phone'] %>",
-  "F[0].Page_1[0].Telephone_Number_LastFourNumbers[1]": "<%= data['surviving_dependent_phone']&.[](6..9) %>",
-  "F[0].Page_1[0].Telephone_Number_SecondThreeNumbers[1]": "<%= data['surviving_dependent_phone']&.[](3..5) %>",
-  "F[0].Page_1[0].Telephone_Number_FirstThreeNumbers[1]": "<%= data['surviving_dependent_phone']&.[](0..2) %>",
-  "F[0].Page_1[0].ClaimantsEMAIL_ADDRESS[0]": "<%= data['surviving_dependent_email']&.[](20..29) %>",
-  "F[0].Page_1[0].ClaimantsEMAIL_ADDRESS[1]": "<%= data['surviving_dependent_email']&.[](0..19) %>",
-  "F[0].Page_1[0].CheckBox1[1]": "<%= data['surviving_dependent_email'] ? 1 : 0 %>",
-  "F[0].Page_1[0].DOB_Month[1]": "<%= data['surviving_dependent_date_of_birth']&.[](5..6) %>",
-  "F[0].Page_1[0].DOB_Day[1]": "<%= data['surviving_dependent_date_of_birth']&.[](8..9) %>",
-  "F[0].Page_1[0].DOB_Year[1]": "<%= data['surviving_dependent_date_of_birth']&.[](0..4) %>",
-  "F[0].Page_1[0].RadioButtonList[2]": "<%= data.dig('veteran_id', 'va_file_number') ? 1 : 2 %>",
+  "F[0].Page_1[0].ClaimantsLastName[0]": "<%= form.data.dig('surviving_dependent_full_name', 'last') %>",
+  "F[0].Page_1[0].ClaimantsMiddleInitial1[0]": "<%= form.data.dig('surviving_dependent_full_name', 'middle') %>",
+  "F[0].Page_1[0].ClaimantsFirstName[0]": "<%= form.data.dig('surviving_dependent_full_name', 'first') %>",
+  "F[0].Page_1[0].ClaimantsSocialSecurityNumber_LastFourNumbers[0]": "<%= form.data.dig('surviving_dependent_id', 'ssn')&.[](5..8) %>",
+  "F[0].Page_1[0].ClaimantsSocialSecurityNumber_SecondTwoNumbers[0]": "<%= form.data.dig('surviving_dependent_id', 'ssn')&.[](3..4) %>",
+  "F[0].Page_1[0].ClaimantsSocialSecurityNumber_FirstThreeNumbers[0]": "<%= form.data.dig('surviving_dependent_id', 'ssn')&.[](0..2) %>",
+  "F[0].Page_1[0].RadioButtonList[0]": "<%= form.data.dig('surviving_dependent_id', 'va_file_number') ? 1 : 2 %>",
+  "F[0].Page_1[0].VA_File_Number[1]": "<%= form.data.dig('surviving_dependent_id', 'va_file_number') || '' %>",
+  "F[0].Page_1[0].RadioButtonList[1]": "<%= form.relationship_to_veteran_radio %>",
```

Comment:
is it ok that this is a function definition rather than a function call?

Reply:
Yeah in Ruby you don't need the parentheses to call the method. If it had `def` in front then it'd be a method definition, I suppose. As a Rubyist-working-in-JS I've been tripped up many a time when I've written `myFunction`, intending to call it not just refer to it, instead of `myFunction()`.